src/core/data.jl

```diff
@@ -717,6 +748,74 @@ function check_transformer_parameters(data::Dict{String,Any})
 end
 
 
+"""
+checks that each storage unit has a reasonable parameters
+"""
+function check_storage_parameters(data::Dict{String,Any})
+    if InfrastructureModels.ismultinetwork(data)
+        error("check_storage_parameters does not yet support multinetwork data")
+    end
+
+    for (i, strg) in data["storage"]
+        if strg["energy"] < 0.0
+            error(LOGGER, "storage unit $(strg["index"]) has a non-positive energy level $(strg["energy"])")
+        end
+        if strg["energy_rating"] < 0.0
+            error(LOGGER, "storage unit $(strg["index"]) has a non-positive energy rating $(strg["energy_rating"])")
+        end
+        if strg["charge_rating"] < 0.0
+            error(LOGGER, "storage unit $(strg["index"]) has a non-positive charge rating $(strg["energy_rating"])")
+        end
+        if strg["discharge_rating"] < 0.0
+            error(LOGGER, "storage unit $(strg["index"]) has a non-positive discharge rating $(strg["energy_rating"])")
+        end
+        if strg["r"] < 0.0
+            error(LOGGER, "storage unit $(strg["index"]) has a non-positive resistance $(strg["r"])")
+        end
+        if strg["x"] < 0.0
+            error(LOGGER, "storage unit $(strg["index"]) has a non-positive reactance $(strg["x"])")
+        end
+        if strg["standby_loss"] < 0.0
+            error(LOGGER, "storage unit $(strg["index"]) has a non-positive standby losses $(strg["standby_loss"])")
+        end
+
+        if haskey(strg, "thermal_rating") && strg["thermal_rating"] < 0.0
+            error(LOGGER, "storage unit $(strg["index"]) has a non-positive thermal rating $(strg["thermal_rating"])")
+        end
+        if haskey(strg, "current_rating") && strg["current_rating"] < 0.0
+            error(LOGGER, "storage unit $(strg["index"]) has a non-positive current rating $(strg["thermal_rating"])")
+        end
+
+        if strg["charge_efficiency"] < 0.0 || strg["charge_efficiency"] > 1.0
+            warn(LOGGER, "storage unit $(strg["index"]) charge efficiency of $(strg["eff_charge"]) is out of the valid range 0.0 to 1.0)")
+        end
+
+        if strg["discharge_efficiency"] < 0.0 || strg["discharge_efficiency"] > 1.0
+            warn(LOGGER, "storage unit $(strg["index"]) discharge efficiency of $(strg["eff_discharge"]) is out of the valid range 0.0 to 1.0)")
+        end
+
+        if !isapprox(strg["x"], 0.0, atol=1e-6, rtol=1e-6)
+            warn(LOGGER, "storage unit $(strg["index"]) has a non-zero reactance $(strg["x"]), which is currently ignored")
+        end
+
+        if strg["charge_rating"] > strg["energy_rating"]
```

Comment:
the charge rating only limits the charge power directly. the interaction with the energy rating depends on the duration of the time steps. e.g. 5 kW charging over at 80% efficiency results in the energy stored increasing by 1 kWh every 0.25 h or 4 kWh every hour. I suggest to either drop this or validate that time_elapsed is strictly larger than 0.

Reply:
Good point, I removed these.